rtlib/rope.hh

```diff
@@ -405,8 +435,57 @@ class Ref {
         append(s);
     }
 
-    template <typename O>
-    void put(O &o) const {
+    void put(std::ofstream &o) const {
```

Comment:
in my head, I thought the `put` methods are for printing. Why do you need a method like this for (de)serialization? Why do you have to overload function with different input arg types? A code comment could help :-)

Reply:
and a second later I scroll down and read your github post about this. Sorry. Adding to the code seems to be worth it!